### ISSUE TITLE
linux golangci-lint v1.49.0, false positive 'unnecessary conversion (unconvert)'

### ISSUE THREAD

**Issue by dev-gto** [golangci/golangci-lint#3214]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc).
- [X] Yes, I've tried with the standalone linter if available. (https://golangci-lint.run/usage/linters/)

### Description of the problem

After migrating to go >= 1.19.0 and golangci-lint >= 1.49.0, the unconvert lint is emitting a false positive while mixing FuncMap from both html/template and web/template (please find code example).

However, the typecast is needed for compilation.

Thanks & regards

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version 1.49.0 built from cc2d97f3 on 2022-08-24T10:24:37Z
```

</details>

### Configuration file

<details>

```console
$ cat .golangci.yml
run:
   deadline: 5m
   tests: false

linters:
   disable-all: true
   enable:
      - govet
      - revive
      - goimports
      - ineffassign
      - stylecheck
      - unconvert
      - typecheck

```

</details>

### Go environment

<details>

```console
$ go version && go env
go version go1.19.1 linux/amd64
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/root/.cache/go-build"
GOENV="/root/.config/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.19.1"
GCCGO="gccgo"
GOAMD64="v1"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/dev/null"
GOWORK=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -Wl,--no-gc-sections -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build1345756025=/tmp/go-build -gno-record-gcc-switches"
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
golangci-lint has version 1.49.0 built from cc2d97f3 on 2022-08-24T10:24:37Z
level=info msg="[config_reader] Config search paths: [./ /go/src/poc /go/src /go / /root]"
level=info msg="[config_reader] Used config file .golangci.yml"
level=info msg="[lintersdb] Active 7 linters: [goimports govet ineffassign revive stylecheck typecheck unconvert]"
level=info msg="[loader] Go packages loading at mode 575 (compiled_files|deps|exports_file|files|name|imports|types_sizes) took 15.444084ms"
level=info msg="[runner/filename_unadjuster] Pre-built 0 adjustments in 15.762µs"
level=error msg="[linters context] typechecking error: pattern ./...: directory prefix . does not contain main module or its selected dependencies"
level=info msg="[linters context/goanalysis] analyzers took 823.678µs with top 10 stages: buildir: 361.039µs, tokenfileanalyzer: 31.06µs, ifaceassert: 27.018µs, unreachable: 25.802µs, cgocall: 17.006µs, ineffassign: 16.955µs, httpresponse: 13.566µs, config: 12.857µs, ST1000: 11.803µs, inspect: 11.197µs"
level=info msg="[runner] processing took 3.459µs with stages: max_same_issues: 706ns, nolint: 324ns, skip_dirs: 285ns, max_from_linter: 284ns, skip_files: 228ns, path_prettifier: 157ns, cgo: 155ns, filename_unadjuster: 155ns, uniq_by_line: 153ns, exclude: 146ns, exclude-rules: 143ns, autogenerated_exclude: 141ns, source_code: 135ns, sort_results: 69ns, path_shortener: 68ns, identifier_marker: 65ns, diff: 64ns, severity-rules: 63ns, max_per_file_from_linter: 60ns, path_prefixer: 58ns"
level=info msg="[runner] linters took 11.090469ms with stages: goanalysis_metalinter: 11.010494ms"
level=info msg="File cache stats: 0 entries of total size 0B"
level=info msg="Memory: 2 samples, avg is 27.4MB, max is 27.4MB"
level=info msg="Execution took 32.513376ms"
main.go:26:28: unnecessary conversion (unconvert)
		Funcs(txttemplate.FuncMap(FuncMap)).
		                         ^

```

</details>

### Code example or link to a public repository

<details>

```go
package main

import (
	"bytes"
	"fmt"
	"html/template"
	txttemplate "text/template"
	"time"
)

var FuncMap = template.FuncMap{
	"ddmmyyyyhhnnss": func(date time.Time) string {
		loc, _ := time.LoadLocation("Europe/Moscow")
		return date.In(loc).Format("02/01/2006 15:04:05")
	},
}

func main() {
	data := struct {
		Date time.Time
	}{
		Date: time.Now().UTC(),
	}

	tmpl, err := txttemplate.New(".").
		Funcs(txttemplate.FuncMap(FuncMap)).
		Parse("{{ddmmyyyyhhnnss .Date}}")

	if err == nil {
		buf := bytes.NewBufferString("")
		err = tmpl.Execute(buf, data)
		if err == nil {
			fmt.Println("[", buf, "]")
		}
	}

	if err != nil {
		fmt.Printf("%s\n", err)
	}
}

```

</details>

**Comment by ldez** [comment]
Hello,

It's not a false positive, `txttemplate.FuncMap` is useless in `Funcs(txttemplate.FuncMap(FuncMap)).`

I can compile with and without the cast.
